### PR TITLE
Config target validation

### PR DESCRIPTION
**Pull request by harveyxia**
BOT NOTES: 
resolves https://github.com/solo-io/service-mesh-hub/issues/962
resolves https://github.com/solo-io/service-mesh-hub/issues/963

### CODE REVIEW

**Comment by joekelley** [comment]
lgtm

File: api/networking/v1alpha2/access_policy.proto [hunk 1]
```diff
@@ -81,4 +81,7 @@ message AccessPolicyStatus {
 
     // The list of Workloads to which this policy has been applied.
     repeated string workloads = 4;
+
+    // A list of errors pertaining to the configuration targets of this resource (e.g. referencing a traffic target that does not exist).
+    repeated string config_target_errors = 5;
```
Comment (ilackarms) [inline api/networking/v1alpha2/access_policy.proto:86]: i don't know that we need to specify `config_target` here. seems like a top level `errors` field would do the trick (and be compatible with any other top-level validation we may do in the future)

File: api/networking/v1alpha2/failover_service.proto [hunk 2]
```diff
@@ -74,6 +74,6 @@ message FailoverServiceStatus {
     // The status of the FailoverService for each Mesh to which it has been applied.
     map<string, ApprovalStatus> meshes = 3;
 
-    // Any errors observed which prevented the resource from being Accepted.
-    repeated string validation_errors = 4;
+    // A list of errors pertaining to the configuration targets of this resource (e.g. referencing a mesh that does not exist).
+    repeated string config_target_errors = 4;
```
Comment (ilackarms) [inline api/networking/v1alpha2/failover_service.proto:78]: same as above

File: api/networking/v1alpha2/traffic_policy.proto [hunk 3]
```diff
@@ -434,4 +434,7 @@ message TrafficPolicyStatus {
 
     // The list of Workloads to which this policy has been applied.
     repeated string workloads = 4;
+
+    // A list of errors pertaining to the configuration targets of this resource (e.g. referencing a traffic target that does not exist).
+    repeated string config_target_errors = 5;
```
Comment (ilackarms) [inline api/networking/v1alpha2/traffic_policy.proto:439]: same as above

File: api/networking/v1alpha2/virtual_mesh.proto [hunk 4]
```diff
@@ -153,8 +153,8 @@ message VirtualMeshStatus {
     // applied to all target meshes.
     ApprovalState state = 2;
 
-    // Any mesh-independent errors observed which prevented the resource from being Accepted.
-    repeated string validation_errors = 3;
+    // A list of errors pertaining to the configuration targets of this resource (e.g. referencing a mesh that does not exist).
+    repeated string config_target_errors = 3;
```
Comment (ilackarms) [inline api/networking/v1alpha2/virtual_mesh.proto:157]: same as above

File: pkg/mesh-networking/apply/applier.go [hunk 5]
```diff
@@ -43,47 +41,53 @@ func (v *applier) Apply(ctx context.Context, input input.Snapshot) {
...
-	for _, trafficPolicy := range input.TrafficPolicies().List() {
+	for _, trafficPolicy := range trafficPolicies {
 		trafficPolicy.Status = networkingv1alpha2.TrafficPolicyStatus{
 			State:              networkingv1alpha2.ApprovalState_ACCEPTED,
 			ObservedGeneration: trafficPolicy.Generation,
 			TrafficTargets:     map[string]*networkingv1alpha2.ApprovalStatus{},
 		}
 	}
-	setWorkloadsForTrafficPolicies(input.TrafficPolicies(), input.Workloads())
+	setWorkloadsForTrafficPolicies(trafficPolicies, workloads)
```
Comment (harveyxia) [inline pkg/mesh-networking/apply/applier.go:77]: Move this to `applyPoliciesToTargets`

**Review by ilackarms — APPROVED** [review]

**Pull request merged**